### PR TITLE
(fix):Added a function to deal with the target species not found in Model

### PR DESCRIPTION
**Pull request by Bensonre**
Went Through 3 iterations of this function think I landed on a pretty good one. Let me know if you want fixes Phill. In short the code calls a function defined in drg.py and imported into PFA and DRGEP at the very beginning of the run_method function that iterates through the Target species and the solution's_object.species().names and check if they are ever equal if so it sets a flag if  the flag is not set at the end of the iteration through solution's_object.species().names the program exits with an error and clear message of what went wrong. 

Ex: 

user runs: 
python __main__.py -m ../example_files/gri30.cti --conditions ../example_files/example_input_file.txt -e 5 --method DRGEP --targets CH4 O2 sk8 --retained_species CH4 O2 N2 CO2 H2O

gets output:
exiting with error target species sk8 not found model

### CODE REVIEW

File: pymars/drg.py [hunk 1]
```diff
@@ -416,6 +417,32 @@ def get_rates_drg(sim_array, solution_object):
         total_edge_data[ic] = ic_edge_data
     return total_edge_data
 
+def target_error_check(solution, target_species):
```
Comment (xMestas) [inline pymars/drg.py]: How about add this function to the helper.py file instead of drg.py

File: pymars/drg.py [hunk 2]
```diff
@@ -416,6 +417,32 @@ def get_rates_drg(sim_array, solution_object):
...
+        networkx graph object of solution
+    target_species : list
+        List of target species to search from
+
+    Returns
+    -------
+    void
+
+    """
+    targetInGraph=False;
```
Comment (xMestas) [inline pymars/drg.py]: Too complex.  Do it like this:

for target in target_species
    if target not in solution.species_names
        sys.exit(...)
return

File: pymars/drg.py [hunk 3]
```diff
@@ -416,6 +417,32 @@ def get_rates_drg(sim_array, solution_object):
         total_edge_data[ic] = ic_edge_data
     return total_edge_data
 
+def target_error_check(solution, target_species):
+    """
+    check if target species are present in graph before attempting to remove them,
```
Comment (xMestas) [inline pymars/drg.py]: This is making sure its in the solution, not in the graph.  Graph hasnt been made yet but if its in the solution it should be added to the graph.

File: pymars/pfa.py [hunk 4]
```diff
@@ -149,6 +149,8 @@ def run_pfa(solution_object, conditions_file, error_limit, target_species, retai
 	if len(target_species) == 0: # If the target species are not specified, puke and die.
 		print("Please specify a target species.")
 		exit()
+	target_error_check(solution_object.species(),target_species)#check if target species is in model
+
```
Comment (xMestas) [inline pymars/pfa.py:153]: Make sure to update the tests corresponding to targets not being in the graph

File: pymars/drg.py (hunk 2 above)
Comment (Bensonre) [reply pymars/drg.py]: oh I wasn't aware of the .Species_names thanks will do

Comment (Bensonre) [reply pymars/drg.py]: .species_names doesnt exsist, so this didnt work. I just change to check the graphs instead of solution obj

**Pull request merged**